app/models/article.rb

```diff
@@ -114,18 +114,55 @@ class Article < ApplicationRecord
   after_commit :sync_related_elasticsearch_docs, on: %i[update]
   after_commit :remove_from_elasticsearch, on: [:destroy]
 
+  # The trigger `update_reading_list_document` is used to keep the `articles.reading_list_document` column updated.
+  #
+  # Its body is inserted in a PostgreSQL trigger function and that joins the columns values
+  # needed to search documents in the context of a "reading list".
+  #
+  # Please refer to https://github.com/jenseng/hair_trigger#usage in case you want to change or update the trigger.
+  #
+  # Additional information on how triggers work can be found in
+  # => https://www.postgresql.org/docs/11/trigger-definition.html
+  # => https://www.cybertec-postgresql.com/en/postgresql-how-to-write-a-trigger/
+  #
+  # Adapted from https://dba.stackexchange.com/a/289361/226575
+  trigger
+    .name(:update_reading_list_document).before(:insert, :update).for_each(:row)
+    .declare("l_org_vector tsvector; l_user_vector tsvector") do
+    <<~SQL
+      NEW.reading_list_document :=
+        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.body_markdown, ''))) ||
+        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_tag_list, ''))) ||
+        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_name, ''))) ||
+        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.cached_user_username, ''))) ||
+        to_tsvector('simple'::regconfig, unaccent(coalesce(NEW.title, ''))) ||
+        to_tsvector('simple'::regconfig,
+          unaccent(
+            coalesce(
+              array_to_string(
+                -- cached_organization is serialized to the DB as a YAML string, we extract only the name attribute
+                regexp_match(NEW.cached_organization, '^name: (.*)$', 'n'),
```

Comment:
While this seems perfectly fine it does make me feel a little bit uneasy. 

![image](https://user-images.githubusercontent.com/47985/113961436-393c7080-9850-11eb-95a9-ae87d84e34e5.png)

Reply:
Actually, are you sure about the `^` here? I ran this via Blazer and for already persisted articles it only works on a subset, whereas removing the anchor works for all articles.

Reply:
Thanks for double checking! I'm going to remove it. It makes me uneasy as well, but unfortunately we serialize a struct instead of saving native JSON.

I could just feed it all to `to_tsvector` but we'd be adding a lot of noise to the column because `cached_organization` contains other fields like the slug and the image URLs